src/colorizer/utils/url_utils.ts

```diff
@@ -2,19 +2,27 @@
 /* global RequestInit */
 
 import { DEFAULT_COLLECTION_FILENAME, DEFAULT_COLLECTION_PATH } from "../../constants";
+import { FeatureThreshold } from "../ColorizeCanvas";
+import { numberToStringDecimal } from "./math_utils";
 
 const URL_PARAM_TRACK = "track";
 const URL_PARAM_DATASET = "dataset";
 const URL_PARAM_FEATURE = "feature";
 const URL_PARAM_TIME = "t";
 const URL_PARAM_COLLECTION = "collection";
+// TODO: Make thresholds/filters language consistent. Requires talking with users/UX!
+const URL_PARAM_THRESHOLDS = "thresholds";
+const URL_PARAM_RANGE = "range";
 
 export type UrlParams = {
-  collection: string | null;
-  dataset: string | null;
-  feature: string | null;
+  collection: string;
+  dataset: string;
+  feature: string;
   track: number;
   time: number;
+  // TODO: bad code smell for url params to be aware of this type
+  thresholds: FeatureThreshold[];
```

Comment:
Is it bad code smell for url_utils.ts to be aware of this type?
Maybe this script shouldn't live in utils?

Reply:
it doesn't seem terrible to me, but this also just feels like a subset of your overall state (guessing since I'm not looking at the rest of the code), so if you wanted to this whole thing could be defined where you're handling your overall app state and then imported here. Otherwise I think it can be nice to have a `types` file for shared types instead of having circular imports.

Reply:
Agreed that it could be good to put this in the `types` file - it's used across lots of files, and it doesn't seem like any single one has a clear claim to being the "owner" of this type.